### PR TITLE
Update FileSystemDialog.xaml.cs

### PR DESCRIPTION
**Pull request by jp-weber**
Fixed issue  #28

### CODE REVIEW

File: SafeExamBrowser.UserInterface.Desktop/Windows/FileSystemDialog.xaml.cs [hunk 1]
```diff
@@ -29,7 +31,6 @@ internal partial class FileSystemDialog : Window
 		private string message;
 		private FileSystemOperation operation;
 		private IWindow parent;
-		private bool restrictNavigation;
```
Comment (dbuechel) [inline SafeExamBrowser.UserInterface.Desktop/Windows/FileSystemDialog.xaml.cs:32]: Why did you remove this flag? We need this for a use case where navigation within the dialog should be restricted to a specific folder.

**Review by dbuechel — CHANGES_REQUESTED** [review]
The build fails. Could you please check why and, if applicable, fix the issue(s).

File: SafeExamBrowser.UserInterface.Desktop/Windows/FileSystemDialog.xaml.cs (hunk 1 above)
Comment (jp-weber) [reply SafeExamBrowser.UserInterface.Desktop/Windows/FileSystemDialog.xaml.cs]: You're right, sorry I didn't see that there were changes in 3.0.1+ and overwrote them. I have reworked this part again.

File: SafeExamBrowser.UserInterface.Desktop/Windows/FileSystemDialog.xaml.cs [hunk 2]
```diff
@@ -286,21 +288,50 @@ private void InitializeDialog()
 			InitializeText();
 			InitializeFileSystem();
 		}
-		
+
 		private DriveInfo[] GetDrives(bool showAll = false)
```
Comment (dbuechel) [inline SafeExamBrowser.UserInterface.Desktop/Windows/FileSystemDialog.xaml.cs:292]: The `showAll` flag isn't really used, correct? If so, I'd remove it to simplify the logic.

**Review by dbuechel — CHANGES_REQUESTED** [review]
You'd also need to implement the same changes for the file system dialog of the mobile UI. Once done, I can approve your pull request.

File: SafeExamBrowser.UserInterface.Desktop/Windows/FileSystemDialog.xaml.cs (hunk 2 above)
Comment (jp-weber) [reply SafeExamBrowser.UserInterface.Desktop/Windows/FileSystemDialog.xaml.cs:292]: Ok I will edit the commit during this week. What I was wondering about the mobile UI part is why the code is not swapped out to a ViewModel so it is less redundant?

Comment (dbuechel) [reply SafeExamBrowser.UserInterface.Desktop/Windows/FileSystemDialog.xaml.cs:292]: Excellent observation. I had to make a decision when designing the architecture, namely whether the user interfaces should share their code or whether they should be completely independent. The former leads to coupling (i.e. changes in one UI affect all the others), the latter leads to code duplication. As I wanted to be able to make a UI that is radically different from others, I chose the latter.

**Review by dbuechel — APPROVED** [review]
Great, thanks!
